rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -913,6 +914,9 @@ RMW_Connext_Publisher::finalize()
   }
 
   if (nullptr != this->topic_mapping) {
+    if (ctx->config.notify_type_disuse(this->topic_mapping->get_net_typecode()) == 0){
+      this->topic_mapping->unregister_plugin_type();
+    }
```

Comment:
Update to use type name instead of type code. Also, we should only check if the mapping has a transformation plugin in it:

```suggestion
    if (nullptr != this->topic_mapping->transform_plugin() &&
      ctx->config.on_network_type_released(this->topic_mapping->network_type_name()) == 0)
    {
      this->topic_mapping->transform_plugin()->unregister_type();
    }
```